win32ss/gdi/eng/stretchblt.c

```diff
@@ -414,6 +641,40 @@ IntEngStretchBlt(SURFOBJ *psoDest,
                             BrushOrigin,
                             Rop4);
     }
+    else
+    {
+        DPRINT("source and dest size are NOT equal.\n");
+    }
+
+    DPRINT("bLeftToRight is '%d' and bTopToBottom is '%d'.\n", bLeftToRight, bTopToBottom);
+
+    /* if cxSrc < 0 then we change the signs for both cxSrc and cxDest and */
+    /* we reverse their coordinates, because these outcomes are the same.  */
+    if (cxSrc < 0)
+    {
+        lTmp = SourceRect->left;
+        SourceRect->left = SourceRect->right;
+        SourceRect->right = lTmp;
+        lTmp = DestRect->left;
+        DestRect->left = DestRect->right;
+        DestRect->right = lTmp;
+        cxSrc = -cxSrc;
+        cxDest = -cxDest;
+    }
+
+    /* if cySrc < 0 then we change the signs for both cySrc and cyDest and */
+    /* we reverse their coordinates, because these outcomes are the same.  */
```

Comment:
```suggestion
    /* if cxSrc < 0 then we change the signs for both cxSrc and cxDest and
     * we reverse their coordinates, because these outcomes are the same. */
    if (cxSrc < 0)
    {
        lTmp = SourceRect->left;
        SourceRect->left = SourceRect->right;
        SourceRect->right = lTmp;
        lTmp = DestRect->left;
        DestRect->left = DestRect->right;
        DestRect->right = lTmp;
        cxSrc = -cxSrc;
        cxDest = -cxDest;
    }

    /* if cySrc < 0 then we change the signs for both cySrc and cyDest and
     * we reverse their coordinates, because these outcomes are the same. */
```